topology/probes/k8s/networkpolicy.go

```diff
@@ -24,173 +24,282 @@ package k8s
 
 import (
 	"fmt"
+	"sync"
 
+	"github.com/skydive-project/skydive/filters"
 	"github.com/skydive-project/skydive/logging"
 	"github.com/skydive-project/skydive/probe"
 	"github.com/skydive-project/skydive/topology/graph"
 
-	api "k8s.io/api/core/v1"
-	networking_v1 "k8s.io/api/extensions/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/client-go/tools/cache"
 )
 
 type networkPolicyProbe struct {
+	sync.Mutex
 	defaultKubeCacheEventHandler
 	graph.DefaultGraphListener
 	*kubeCache
-	graph      *graph.Graph
-	podCache   *kubeCache
-	podIndexer *graph.MetadataIndexer
+	graph          *graph.Graph
+	podCache       *kubeCache
+	namespaceCache *kubeCache
+	objIndexer     *graph.MetadataIndexer
 }
 
-func (n *networkPolicyProbe) newMetadata(np *networking_v1.NetworkPolicy) graph.Metadata {
-	return newMetadata("networkpolicy", np.GetNamespace(), np.GetName(), np)
+func newObjectIndexerByNetworkPolicy(g *graph.Graph) *graph.MetadataIndexer {
+	ownedByFilter := filters.NewOrFilter(
+		filters.NewTermStringFilter("Type", "namespace"),
+		filters.NewTermStringFilter("Type", "pod"),
+	)
+
+	filter := filters.NewAndFilter(
+		filters.NewTermStringFilter("Manager", managerValue),
+		ownedByFilter,
+	)
+	m := graph.NewGraphElementFilter(filter)
+	return graph.NewMetadataIndexer(g, m)
+}
+
+func (n *networkPolicyProbe) newMetadata(np *v1beta1.NetworkPolicy) graph.Metadata {
+	return newMetadata("networkpolicy", np.Namespace, np.Name, np)
 }
 
-func networkPolicyUID(np *networking_v1.NetworkPolicy) graph.Identifier {
+func networkPolicyUID(np *v1beta1.NetworkPolicy) graph.Identifier {
 	return graph.Identifier(np.GetUID())
 }
 
-func dumpNetworkPolicy(np *networking_v1.NetworkPolicy) string {
-	return fmt.Sprintf("networkPolicy{Namespace: %s Name: %s}", np.GetNamespace(), np.GetName())
+func dumpNetworkPolicy(np *v1beta1.NetworkPolicy) string {
+	return fmt.Sprintf("networkPolicy{Namespace: %s, Name: %s}", np.Namespace, np.Name)
 }
 
 func (n *networkPolicyProbe) OnAdd(obj interface{}) {
-	if policy, ok := obj.(*networking_v1.NetworkPolicy); ok {
+	n.Lock()
+	defer n.Unlock()
+	if np, ok := obj.(*v1beta1.NetworkPolicy); ok {
```

Comment:
Please stop renaming local variables

Reply:
Other parts of the file use `np` for referring to a `v1.NetworkPolicy` the change was done for consistency throughout the file.